### PR TITLE
deck file kong2kic docs

### PR DESCRIPTION
**Pull request by battlebyte**
### Description

I added the documentation for the upcoming deck file `kong2kic` subcommand. I updated as well the landing page for the `deck` command.
 

https://konghq.atlassian.net/browse/APIOPS-8

### Testing instructions

landing page: https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/
kongair page: https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/use-case/
kon2kic: https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/reference/deck_file_kong2kic/

### Checklist 

- [review:sme] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

### CODE REVIEW

**Comment by mheap** [comment]
@lena-larionova Could you review this PR at some point this week? I'd like to trigger a decK release with this functionality

**Comment by lena-larionova** [comment]
@mheap did this not go into the 1.34 release that went out last week? So we assume there's a 1.35 coming with this feature?

**Comment by mheap** [comment]
Yes, 1.35 coming soon. Likely sometime next week

**Comment by lena-larionova** [comment]
@battlebyte I moved your new page into 1.35.x, since it's not available in 1.30 (did it directly cause that was simpler, and needed doing either way 😄 ). I'll do the rest of the review on the PR.

File: app/_src/deck/index.md [hunk 1]
```diff
@@ -1,72 +1,137 @@
...
+   - [deck file add-plugins](/deck/{{page.release}}/reference/deck_file_add-plugins/): Incorporates plugins into decK objects.
+   - [deck file add-tags](/deck/{{page.release}}/reference/deck_file_add-tags/): Attaches tags to decK objects for enhanced organization.
+   - [deck file list-tags](/deck/{{page.release}}/reference/deck_file_list-tags/): Enumerates tags associated with decK objects.
+   - [deck file remove-tags](/deck/{{page.release}}/reference/deck_file_remove-tags/): Eliminates tags from decK objects.
+   - [deck file lint](/deck/{{page.release}}/reference/deck_file_lint/): Assesses a decK file against a set of linting rules, identifying any discrepancies.
+   - [deck file patch](/deck/{{page.release}}/reference/deck_file_patch/): Applies modifications to a decK file without overhauling the entire configuration.
+   - [deck file merge](/deck/{{page.release}}/reference/deck_file_merge/): Combines several partial decK files into a comprehensive one.
+   - [deck file render](/deck/{{page.release}}/reference/deck_file_render/): Fuses multiple complete decK files, rendering a singular, unified configuration.
```
Comment (lena-larionova) [inline app/_src/deck/index.md:41]:
```suggestion
   - [deck file namespace](/deck/{{page.release}}/reference/deck_file_namespace) - Apply a namespace to routes in a decK file by prefixing the path.
   - [deck file render](/deck/{{page.release}}/reference/deck_file_render/): Fuses multiple complete decK files, rendering a singular, unified configuration.
```
Command was added in 1.34

**Comment by lena-larionova** [comment]
Since you're reworking the landing page, I think it's worth redoing the whole thing. 
I took a stab at restructuring the rest: https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/

Do we also want to add some quick links to the top of the page, like this? 
https://docs.konghq.com/gateway-operator/latest/

Then we could move the KongAir example to its own page (it's a bit overwhelming on the intro) and have a quick link directly to it.

**Comment by battlebyte** [comment]
Thanks for the changes @lena-larionova ! Agree with your suggestion, quick links at the top of the page will help in structuring it better.

**Comment by lena-larionova** [comment]
Here we go: 
* landing page with quick links: https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/
* kongair page (minor tweak with the validate section to throw into tabs, for readability): https://deploy-preview-6905--kongdocs.netlify.app/deck/unreleased/use-case/

File: app/_src/deck/index.md [hunk 2]
```diff
@@ -1,106 +1,117 @@
...
-commands.
+{% if_version gte:1.35.x %}
+## decK commands
+
+The decK commands are structured into three main categories:
+
+1. **Configuration Generation**: This category focuses on the initial creation of decK state files from industry-standard API specification formats. This category includes the following command: 
+   - [deck file openapi2kong](/deck/{{page.release}}/reference/deck_file_openapi2kong/): Converts OpenAPI specification files into decK state files.
+
+2. **Configuration Transformation**: This set of commands provides the tools needed to refine and restructure decK configuration files. It allows for the segmentation of a full configuration into smaller, manageable parts and their subsequent reassembly. This category includes commands such as:
```
Comment (lena-larionova) [inline app/_src/deck/index.md:66]:
```suggestion
2. **Configuration Transformation**: This set of commands provides the tools needed to refine and restructure decK configuration files. It allows for the segmentation of a full configuration into smaller, manageable parts and their subsequent reassembly. This category includes the following commands:
```
"such as" implies that there are more

File: app/_src/deck/index.md [hunk 3]
```diff
@@ -1,106 +1,117 @@
...
+   - [deck file add-plugins](/deck/{{page.release}}/reference/deck_file_add-plugins/): Incorporates plugins into decK objects.
+   - [deck file add-tags](/deck/{{page.release}}/reference/deck_file_add-tags/): Attaches tags to decK objects for enhanced organization.
+   - [deck file list-tags](/deck/{{page.release}}/reference/deck_file_list-tags/): Enumerates tags associated with decK objects.
+   - [deck file remove-tags](/deck/{{page.release}}/reference/deck_file_remove-tags/): Eliminates tags from decK objects.
+   - [deck file lint](/deck/{{page.release}}/reference/deck_file_lint/): Assesses a decK file against a set of linting rules, identifying any discrepancies.
+   - [deck file patch](/deck/{{page.release}}/reference/deck_file_patch/): Applies modifications to a decK file without overhauling the entire configuration.
+   - [deck file merge](/deck/{{page.release}}/reference/deck_file_merge/): Combines several partial decK files into a comprehensive one.
+   - [deck file namespace](/deck/{{page.release}}/reference/deck_file_namespace) - Apply a namespace to routes in a decK file by prefixing the path.
```
Comment (lena-larionova) [inline app/_src/deck/index.md:74]:
```suggestion
   - [deck file namespace](/deck/{{page.release}}/reference/deck_file_namespace): Apply a namespace to routes in a decK file by prefixing the path.
```

File: app/_src/deck/index.md [hunk 4]
```diff
@@ -1,106 +1,117 @@
...
+   - [deck file render](/deck/{{page.release}}/reference/deck_file_render/): Fuses multiple complete decK files, rendering a singular, unified configuration.
+   - [deck file validate](/deck/{{page.release}}/reference/deck_file_validate/): Conducts an offline validation of the state file, pinpointing potential issues.
+   - [deck file kong2kic](/deck/{{page.release}}/reference/deck_file_kong2kic/): Converts decK state files into {{site.kic_product_name}} manifests.
+
+3. **Gateway State Management**: This category encompasses commands that facilitate
+the synchronization of the final decK file with the target platform, be it {{site.konnect_product_name}}, {{site.base_gateway}}, 
+or {{site.kic_product_name}}. Key commands include:
```
Comment (lena-larionova) [inline app/_src/deck/index.md:81]:
```suggestion
or {{site.kic_product_name}}. This category includes:
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 5]
```diff
@@ -0,0 +1,734 @@
+---
+title: deck file kong2kic
+source_url: https://github.com/Kong/deck/tree/main/cmd/file_kong2kic.go
+content_type: reference
+---
+
+Convert decK state files to {{site.kic_product_name}} kubernetes manifests.
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:7]:
```suggestion
```
This is a duplicate of the next line.

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 6]
```diff
@@ -0,0 +1,734 @@
...
+
+The `kong2kic` subcommand transforms Kong's configuration files, written in the deck format, into Kubernetes manifests suitable for the {{site.kic_product_name}}. This tool serves as a bridge for deploying API configurations from deck files directly to a Kubernetes cluster running {{site.kic_product_name}}, facilitating  integration into the final stages of an APIOps pipeline. Essentially, `kong2kic` translates the desired API configurations into a format that `kubectl` can deploy, ensuring that the API's intended state is accurately reflected in the Kubernetes environment.
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:9]:
```suggestion
The `kong2kic` subcommand transforms Kong's configuration files, written in the decK format, into Kubernetes manifests suitable for the {{site.kic_product_name}}. This tool serves as a bridge for deploying API configurations from decK files directly to a Kubernetes cluster running {{site.kic_product_name}}, facilitating integration into the final stages of an APIOps pipeline. Essentially, `kong2kic` translates the desired API configurations into a format that `kubectl` can deploy, ensuring that the API's intended state is accurately reflected in the Kubernetes environment.
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 7]
```diff
@@ -0,0 +1,734 @@
...
+
+`kong2kic` integrates API management within kubernetes environments, offering support for both Gateway API and Ingress API standards.
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:11]:
```suggestion
`kong2kic` integrates API management within Kubernetes environments, offering support for both Gateway API and Ingress API standards.
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 8]
```diff
@@ -0,0 +1,734 @@
...
+Currently supports the HTTP/HTTPS protocol.
+
+The following table details how Kong configuration entities will be mapped to Kubernetes manifests:
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:14]:
```suggestion
The following table details how Kong configuration entities are mapped to Kubernetes manifests:
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 9]
```diff
@@ -0,0 +1,734 @@
...
+
+| decK entity                                                  | K8s entity                                                        |
+|--------------------------------------------------------------|-------------------------------------------------------------------|
+| Service                                                      | Service with annotations and KongIngress for upstream section     |
+| Route                                                        | Ingress (Ingress API) or HTTPRoute (Gateway API) with annotations |
+| Global Plugin                                                | KongClusterPlugin                                                 |
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:20]:
```suggestion
| Global plugin                                                | KongClusterPlugin                                                 |
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 10]
```diff
@@ -0,0 +1,734 @@
...
+| Plugin                                                       | KongPlugin                                                        |
+| Auth Plugins <br>(`key-auth`, `hmac-auth`, `jwt`, `basic-auth`, `oauth2`, `acl`, `mtls-auth`) | KongPlugin and Secret with credentials section in KongConsumer    |
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:22]:
```suggestion
| Auth plugins <br>(`key-auth`, `hmac-auth`, `jwt`, `basic-auth`, `oauth2`, `acl`, `mtls-auth`) | KongPlugin and Secret with credentials section in KongConsumer    |
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 11]
```diff
@@ -0,0 +1,734 @@
...
+| Upstream                                                     | KongIngress or kongUpstreamPolicy                                 |
+| Consumer                                                     | KongConsumer                                                      |
+| ConsumerGroup                                                | KongConsumerGroup                                                 |
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:25]:
```suggestion
| Consumer group                                                | KongConsumerGroup                                                 |
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 12]
```diff
@@ -0,0 +1,734 @@
...
+          namespace: example_namespace
+          strategy: local
+          hide_client_headers: false
+          window_type: sliding
+          retry_after_jitter_max:
+          - 0
+```
+
+{% endnavtab %}
+{% navtab Convert to Gateway API %}
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:150]:
```suggestion
{% navtab Converted to Gateway API %}
```

File: app/_src/deck/use-case.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
+---
+title: decK Use Case - Streamlining KongAir APIs
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:2]:
```suggestion
title: decK Use Case - Streamlining KongAir API Management
```
Does this other title make more sense? Would need to adjust the nav file as well, if that's the case.

File: app/_src/deck/use-case.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+---
+
+Let's explore how KongAir, an imaginary airline, leverages decK to streamline its API management processes. The KongAir API Community of Practice has established a set of governance rules to ensure uniformity and efficiency across all API teams:
+
+- Every API team within KongAir, including those responsible for the Flights and Routes APIs, adopts OpenAPI specifications to define their API contracts.
+- These teams maintain the flexibility to employ Kong's Transformation and Validation plugins to enhance their APIs. They manage these plugins' configurations through modular decK state files, promoting autonomy and customization.
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:8]:
```suggestion
- These teams maintain the flexibility to employ Kong's transformation and validation plugins to enhance their APIs. They manage these plugins' configurations through modular decK state files, promoting autonomy and customization.
```

File: app/_src/deck/use-case.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+- The KongAir API Community of Practice has also embraced a comprehensive set of API Design Guidelines to standardize API development. These guidelines are implemented and monitored through a linting file overseen by the dedicated API Platform Team, ensuring adherence to best practices.
+- The API Platform Team assumes a pivotal role in configuring critical plugins related to Observability, Security, and Traffic Control within Kong, centralizing expertise and governance for these essential aspects.
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:10]:
```suggestion
- The API Platform Team assumes a pivotal role in configuring critical plugins related to observability, security, and traffic control within Kong, centralizing expertise and governance for these essential aspects.
```

File: app/_src/deck/reference/deck_file_kong2kic.md [hunk 16]
```diff
@@ -0,0 +1,734 @@
...
+kind: KongConsumerGroup
+metadata:
+  annotations:
+    kubernetes.io/ingress.class: kong
+  name: example-consumer-group
+---
+```
+
+{% endnavtab %}
+{% navtab Convert to Ingress API %}
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_kong2kic.md:450]:
```suggestion
{% navtab Converted to Ingress API %}
```

File: app/_src/deck/use-case.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+- Furthermore, this team is tasked with the management of environment-specific variables, ensuring seamless deployment and operation across different stages of the development lifecycle.
+
+The diagram below delineates KongAir's structured approach to deploying decK, steered by their established governance protocols:
+
+1. The Flights API team initiates the process by converting their OpenAPI Specification into a decK state file using the `deck file openapi2kong` command.
+2. Next, they enhance the state file by integrating Transformation Plugins (such as Request Transformer Advanced and Correlation ID) and Validation plugins (like OAS Validation) using the `deck file add-plugins` command.
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:16]:
```suggestion
2. Next, they enhance the state file by integrating transformation plugins (such as [Request Transformer Advanced](/hub/kong-inc/request-transformer-advanced/) and [Correlation ID](/hub/kong-inc/correlation-id/) and validation plugins (like [OAS Validation](/hub/kong-inc/oas-validation/) using the `deck file add-plugins` command.
```
Might as well use this opportunity to link to plugins :)

File: app/_src/deck/use-case.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+3. To track the configuration's creation time, they apply relevant tags using `deck file add-tags`.
+4. The state file undergoes a quality check against a predefined linting ruleset with `deck file lint`, ensuring adherence to best practices.
+5. Environment-specific adjustments, including upstream API URLs, are made using the `deck file patch` command.
+6. The Platform Team then merges global plugins for Observability, Authentication, Authorization, and Traffic Control into the main state file with `deck file merge`.
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:20]:
```suggestion
6. The Platform Team then merges global plugins for observability, authentication, authorization, and traffic control into the main state file with `deck file merge`.
```

File: app/_src/deck/use-case.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+7. At this stage, a comprehensive state file for the Flights API is ready. This file is combined with the Routes API's state file using `deck file render`, creating a unified configuration.
+8. The final state file is subjected to an offline validation through `deck file validate`.
+
+{% capture validate %}
+{% navtabs %}
+{% navtab Konnect or Gateway %}
+For deployments to {{site.konnect_product_name}} or {{site.base_gateway}} deployments, the process involves:
```
Comment (lena-larionova) [inline app/_src/deck/use-case.md:27]:
```suggestion
For {{site.konnect_product_name}} or {{site.base_gateway}} deployments, the process involves:
```

**Review by lena-larionova — COMMENTED** [review]
A bunch of minor edits, otherwise LGTM!

pinging @mheap if you need to merge Monday (I'll be out).

File: app/_src/deck/index.md [hunk 20]
```diff
@@ -1,106 +1,117 @@
 ---
 title: decK
-subtitle: Manage Konnect and Kong Gateway configuration declaratively
+subtitle: Manage {{site.konnect_product_name}}, {{site.base_gateway}} and {{site.kic_product_name}} configuration declaratively
```
Comment (lena-larionova) [inline app/_src/deck/index.md:3]:
```suggestion
subtitle: Manage {{site.konnect_product_name}}, {{site.base_gateway}}, and {{site.kic_product_name}} configuration declaratively
```

**Comment by battlebyte** [comment]
Looks great @lena-larionova , thanks for your help. I'm not sure if we provide links in our docs to github repositories. There is one for KongAir that implements some of the features explained in the documentation: https://github.com/Kong/KongAir. Is it ok to add that link to the KongAir use case page?

**Comment by lena-larionova** [comment]
@battlebyte sorry, missed your last comment. Yes, we can add links to repos, as long as they're public. So there's no issue linking to the KongAir repo.

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**